src/data/set/basic.lean

```diff
@@ -1440,10 +1440,10 @@ def range (f : ι → α) : set α := {x | ∃y, f y = x}
 
 @[simp] theorem mem_range_self (i : ι) : f i ∈ range f := ⟨i, rfl⟩
 
-theorem forall_range_iff {p : α → Prop} : (∀ a ∈ range f, p a) ↔ (∀ i, p (f i)) :=
+@[simp] theorem forall_range_iff {p : α → Prop} : (∀ a ∈ range f, p a) ↔ (∀ i, p (f i)) :=
 ⟨assume h i, h (f i) (mem_range_self _), assume h a ⟨i, (hi : f i = a)⟩, hi ▸ h i⟩
```

Comment:
I think I can fix this with
```suggestion
@[simp] theorem forall_range_iff_aux {p : α → Prop} : (∀ a i, f i = a → p a) ↔ (∀ i, p (f i)) :=
⟨assume h i, h (f i) i rfl, assume h a i hi, hi ▸ h i⟩

theorem forall_range_iff {p : α → Prop} : (∀ a ∈ range f, p a) ↔ (∀ i, p (f i)) :=
by simp
```
What would be a better name for `forall_range_iff_aux`?

Reply:
The best I can come up with is `forall_apply_eq_imp_iff`. And it certainly is a *good* `simp` lemma :grinning:

Reply:
If if helps, parts of it look like `forall_eq'`, and the lemma used to prove `exists_range_iff` is `exists_exists_eq_and`.

Reply:
I'm also happy with `forall_range_iff'`. Whatever you want.

Reply:
Alright, added using your suggested name.